src/charm.py

```diff
@@ -1281,18 +1283,39 @@ def _get_runner_scaler(
         )
         reactive_runner_config = None
         if reactive_config := state.reactive_config:
+            # The charm is not able to determine which architecture the runner is running on,
+            # so we add all architectures to the supported labels.
+            supported_labels = set(self._create_labels(state)) | GITHUB_SELF_HOSTED_ARCH_LABELS
```

Comment:
Isn't that going to be a problem? How would it know that it can pick up `edge` for `x64` but not for `arm64`?

Reply:
This is not currently a problem, as we are assuming a different queue for each flavour in the first iteration of the reactive feature. This means that the x64/arm64 messages would be routed to a different queue, and the charms supporting arm64 would only connect to the queue for arm64. The label validation here is currently only there to try and catch programming or routing errors (but you are right that it wouldn't actually be able to detect those errors for the architectures).

Once we allow the charm to spawn different flavours (e.g. with custom images) this could become a problem, as we could end up with a queue of messages containing different architectures.